src/main/java/frc/robot/subsystems/swerve/states/aimassist/AimAssistMath.java

```diff
@@ -0,0 +1,80 @@
+package frc.robot.subsystems.swerve.states.aimassist;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import frc.robot.subsystems.swerve.SwerveConstants;
+import frc.robot.subsystems.swerve.SwerveMath;
+import frc.robot.subsystems.swerve.states.SwerveState;
+
+public class AimAssistMath {
+
+	public static ChassisSpeeds getRotationAssistedChassisSpeeds(
+		ChassisSpeeds speeds,
+		Rotation2d robotHeading,
+		Rotation2d targetHeading,
+		SwerveConstants swerveConstants
+	) {
+		Rotation2d pidOutputVelocityPerSecond = Rotation2d
+			.fromDegrees(swerveConstants.rotationDegreesPIDController().calculate(robotHeading.getDegrees(), targetHeading.getDegrees()));
+
+		Rotation2d rotationalVelocityPerSecond = applyMagnitudeCompensation(pidOutputVelocityPerSecond, SwerveMath.getDriveMagnitude(speeds));
+		Rotation2d combinedRotationalVelocityPerSecond = Rotation2d
+			.fromRadians(rotationalVelocityPerSecond.getRadians() + speeds.omegaRadiansPerSecond);
+		Rotation2d clampedRotationalVelocityPerSecond = SwerveMath
+			.clampRotationalVelocity(combinedRotationalVelocityPerSecond, swerveConstants.maxRotationalVelocityPerSecond());
+
+		return new ChassisSpeeds(speeds.vxMetersPerSecond, speeds.vyMetersPerSecond, clampedRotationalVelocityPerSecond.getRadians());
+	}
+
+	/**
+	 * @formatter:off
+	 * Returns {@link ChassisSpeeds} that aligns you to do object.
+	 * The returned chassis speeds will move you horizontally to the object so your current heading will point to it.
+	 * Example (0 is object, R is robot, > is heading):
+	 * Current Pose:              Ending Pose:
+	 * |            0          |   |   R>       0          |
+	 * |                       |   |                       |
+	 * |   R>                  |   |                       |
+	 * |                       |   |                       |
+	 * |                       |   |                       |
+	 * |                       |   |                       |
```

Comment:
```suggestion
	 * Returns {@link ChassisSpeeds} that aligns you to do object.
	 * The returned chassis speeds will move you horizontally to the object so your current heading will point to it.
	 * Example (0 is object, R is robot, > is heading):
	 * Current Pose:                Ending Pose:
	 * |            0          |   |   R>       0          |
	 * |   ^                   |   |                       |
	 * |   R                   |   |                       |
	 * |                       |   |                       |
```

1. you don't need those extra lines
2. i think it's a bit more clear that the heading changes when it's part of the ascii visual too
3. horizontally relative to what?

Reply:
horizontal to the object, makes your heading point to ther object by moving your x y and not changing heading

Reply:
![image](https://github.com/user-attachments/assets/d413d3cb-113d-4ac5-9df5-3c01e44ba12a)
what does it do in this situation?

Reply:
u move up

Reply:
i meant that in this situation its impossible to be in the object heading because its in the corner